src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -2585,9 +2585,7 @@ codet java_bytecode_convert_methodt::convert_instructions(
     for(const auto &exception_row : method.exception_table)
     {
       // add the CATCH-POP before the end of the try block
-      if(cur_pc<exception_row.end_pc &&
-         !working_set.empty() &&
-         *working_set.begin()==exception_row.end_pc)
+      if(cur_pc==exception_row.end_pc)
```

Comment:
Is there a test to cover these infinite loops?

Reply:
And if so can it be put in the appropriate commit :slightly_smiling_face:

Reply:
exception27 cover this case

Reply:
Then exception27 is in the wrong commit.

Reply:
Exception27 covers a number of cases simultaneously. It will fail however without the second commit.
Personally, I'd rather not increase the CI duration by adding redundant tests.

Reply:
May I challenge this: You will know about Delta Debugging, and focussed regression tests are the optimal outcome of that process. Tests that fail for a multitude of reasons make this unnecessarily hard.

I'd thus ask to either split Exception27 in two (each covering one of the two commits) if feasible, or else come up with another one that exclusively covers the infinite loops.

Should it be infeasible to come up with a test that only covers infinite loops, then maybe the two commits shouldn't be split in the first place.